greenlight/lib/tests/test_tabs.py

```diff
@@ -47,9 +47,7 @@ def tearDown(self):
             gBrowser.removeAllTabsBut(gBrowser.tabs[0]);
         """)
 
-        # Needs to be set by the harness itself
-        self.prefs.restore_pref('browser.tabs.warnOnClose')
-        self.prefs.restore_pref('browser.tabs.warnOnCloseOtherTabs')
+        self.wait_for_condition(lambda _: len(self.browser.tabbar.tabs) == 1)
```

Comment:
This is not really related to the topic of this PR. Have you seen any failures? The whole code needs a rewrite anyhow.